engine/table/src/main/java/io/deephaven/engine/table/impl/select/MatchFilter.java

```diff
@@ -46,21 +47,16 @@ static MatchFilter ofLiterals(
             Collection<Literal> literals,
             boolean inverted) {
         return new MatchFilter(
-                null, null, null,
                 inverted ? MatchType.Inverted : MatchType.Regular,
                 columnName,
                 literals.stream().map(AsObject::of).toArray());
     }
 
-    /** The expression to use for this filter should initialization fail. */
-    private final String expression;
-    /** The failover filter to use if this filter fails to initialize. */
-    private WhereFilter failoverFilter;
-    /** The parser configuration to use for the failover filter. */
-    private final FormulaParserConfiguration parserConfiguration;
+    /** A fail-over where filter supplier should the match filter initialization fail. */
+    private final CachingSupplier<WhereFilter> failoverFilter;
 
     @NotNull
-    private final String columnName;
+    private String columnName;
```

Comment:
There's an accessor for this. Is it still safe? Might be sufficient to JavaDoc accessors as only safe after init, and then verify usages.